From bbc7249dc092563e9b014e6436b8a5970461a0f8 Mon Sep 17 00:00:00 2001
From: Miranda Kephart <mkephart@google.com>
Date: Mon, 8 Feb 2021 11:30:34 -0500
Subject: [PATCH] Make screenshot error notification PendingIntent immutable

Bug: 178189250
Test: manual (using POC in bug)
Change-Id: I787915c0493d18500e7531bd0ca7191532441f93
(cherry picked from commit 1eecc874b6bda42a5a191684704e3bf424cf47da)
---
 .../systemui/screenshot/ScreenshotNotificationsController.java  | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/packages/SystemUI/src/com/android/systemui/screenshot/ScreenshotNotificationsController.java b/packages/SystemUI/src/com/android/systemui/screenshot/ScreenshotNotificationsController.java
index 42dde4064a97..b8a468e0ce45 100644
--- a/packages/SystemUI/src/com/android/systemui/screenshot/ScreenshotNotificationsController.java
+++ b/packages/SystemUI/src/com/android/systemui/screenshot/ScreenshotNotificationsController.java
@@ -252,7 +252,7 @@ public class ScreenshotNotificationsController {
                 dpm.createAdminSupportIntent(DevicePolicyManager.POLICY_DISABLE_SCREEN_CAPTURE);
         if (intent != null) {
             final PendingIntent pendingIntent = PendingIntent.getActivityAsUser(
-                    mContext, 0, intent, 0, null, UserHandle.CURRENT);
+                    mContext, 0, intent, PendingIntent.FLAG_IMMUTABLE, null, UserHandle.CURRENT);
             b.setContentIntent(pendingIntent);
         }

--
2.29.0
